### PR TITLE
Request to adding UML diagram #2

### PR DESCRIPTION
**Pull request by ghost**
- Re-created UML diagram  c29fe57
- Added UML diagram classes  fcad997
- Fixed some UML diagram classes  49337ec
- Fixed some UML classes  638fb80
- Fixed UML diagram  328b396
- UML architecture fixes  99f7439
- Fixed UML diagram  cf28cde
- UML diagram architecture  e6b1310
- Some fixes in UML  4edd0fb
- Small fix in UML diagram  b35cf8c

### CODE REVIEW

**Comment by ghost** [comment]
This is a new pull request. Request #9 is no longer relevant.

File: src/main/resources/diagram_git_starship.puml [hunk 1]
```diff
@@ -2,132 +2,187 @@
 
 package rape.brutal.gitstarship {
 
-    package ships {
+    class Universe {
+        -int kills;
+        -int moneySpends;
+        +void startUniverse();
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml]: Maybe `journeyState()`?

File: src/main/resources/diagram_git_starship.puml [hunk 2]
```diff
@@ -2,132 +2,187 @@
...
-            -Turret leftTurret;
-        }
+        +abstract class Event {
+            #Universe universe;
+            #Location location;
+            #GitStarship gitStarship;
 
-        class PirateStarship {
-            -Turret rightTurret;
+            +void Event(Universe universe, Location location, GitStarship gitStarship);
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml]: Event may not contain a 'location'.

File: src/main/resources/diagram_git_starship.puml [hunk 3]
```diff
@@ -2,132 +2,187 @@
...
+            +void start(GitStarship gitStarship);
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml]: If `gitStarship` assigned via the constructor, it is a unnesessary to pass it through `start()` parameter.

File: src/main/resources/diagram_git_starship.puml [hunk 4]
```diff
@@ -2,132 +2,187 @@
...
 
-        Trader -o Planet
-        SpacePort o-- Trader
-        Trader --o Asteroid
-        Caravan o- Trader
+        Starship <|-- GitStarship
+        Starship <|-- PirateStarship
 
-        class Planet {
+        abstract class Starship implements IAttackable {
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml:88]: Your ships can't land to 'ILandable'.

File: src/main/resources/diagram_git_starship.puml [hunk 5]
```diff
@@ -2,132 +2,187 @@
...
+        +class Asteroid extends Location  implements ILandable {
+            -Trader trader;
+            -Storage storage;
         }
 
-        abstract class ValuableItem {
-            -int value;
+        +class OpenSpace extends Location {
+            -ArrayList<Starship> ships;
+            -Storage storage;
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml]: Now, if all subclasses of `Location` has `storage`, why `storage` is not presented in base class?

File: src/main/resources/diagram_git_starship.puml [hunk 6]
```diff
@@ -2,132 +2,187 @@
...
-    interface ILander {
-        +void landTo(ILandable iLandable);
-        +void takeOff();
     }
 
-    class Storage {
-        -HashMap<String, IStoreable> iStoreableHashMap;
+    package stuff {
+
+        +class MedicineChest implements IStorageable {
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml]: How this class works?

**Review by ViRGiL175 — CHANGES_REQUESTED** [review]

File: src/main/resources/diagram_git_starship.puml (hunk 1 above)
Comment (ghost) [reply src/main/resources/diagram_git_starship.puml]: Fixed!

File: src/main/resources/diagram_git_starship.puml (hunk 2 above)
Comment (ghost) [reply src/main/resources/diagram_git_starship.puml]: Fixed!

File: src/main/resources/diagram_git_starship.puml (hunk 3 above)
Comment (ghost) [reply src/main/resources/diagram_git_starship.puml]: Fixed!

File: src/main/resources/diagram_git_starship.puml (hunk 5 above)
Comment (ghost) [reply src/main/resources/diagram_git_starship.puml]: Fixed!

File: src/main/resources/diagram_git_starship.puml (hunk 4 above)
Comment (ghost) [reply src/main/resources/diagram_git_starship.puml:88]: Fixed!

File: src/main/resources/diagram_git_starship.puml (hunk 6 above)
Comment (ghost) [reply src/main/resources/diagram_git_starship.puml]: Fixed! Delete this package.

**Comment by ghost** [comment]
Ok, I add some fixes in UML! please review again!

File: src/main/resources/diagram_git_starship.puml [hunk 7]
```diff
@@ -2,132 +2,174 @@
...
+            +void start();
         }
 
     }
 
-    package stuff {
-
-        ValuableItem <|-- TurretMagazineCase
-        ValuableItem <|-- MetalShell
+    interface ILandable {
```
Comment (ViRGiL175) [inline src/main/resources/diagram_git_starship.puml:53]: Unclear interface using.

**Review by ViRGiL175 — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
I has made some fixes in UML! please review again!

**Review by ViRGiL175 — APPROVED** [review]

**Comment by ViRGiL175** [comment]
Approved.

**Pull request merged**